src/ai/backend/manager/models/vfolder.py

```diff
@@ -1210,6 +1210,172 @@ async def batch_load_by_user(
+            .limit(limit)
+            .offset(offset)
+        )
+        if filter is not None:
+            qfparser = QueryFilterParser(cls._queryfilter_fieldspec)
+            query = qfparser.append_filter(query, filter)
+        if order is not None:
+            qoparser = QueryOrderParser(cls._queryorder_colmap)
+            query = qoparser.append_ordering(query, order)
+        else:
+            query = query.order_by(vfolders.c.created_at.desc())
+        async with graph_ctx.db.begin_readonly() as conn:
+            return [
+                obj
+                async for r in (await conn.stream(query))
+                if (obj := cls.from_row(graph_ctx, r)) is not None
+            ]
+
+    @classmethod
+    async def load_count_project(
+        cls,
+        graph_ctx: GraphQueryContext,
+        *,
+        domain_name: str = None,
+        group_id: uuid.UUID = None,
+        user_id: uuid.UUID = None,
+        filter: str = None,
+    ) -> int:
+        from ai.backend.manager.models import association_groups_users as agus
+
+        from .group import groups
+        from .user import users
+
+        j = sa.join(agus, users, agus.c.user_id == users.c.uuid)
+        query = sa.select([agus.c.group_id]).select_from(j).where(agus.c.user_id == user_id)
+
+        async with graph_ctx.db.begin_readonly() as conn:
+            result = await conn.execute(query)
+
+        grps = result.fetchall()
+        group_ids = [g.group_id for g in grps]
+        j = vfolders.join(groups, vfolders.c.group == groups.c.id)
+        query = sa.select([sa.func.count()]).select_from(j).where(vfolders.c.group.in_(group_ids))
+
+        if filter is not None:
+            qfparser = QueryFilterParser(cls._queryfilter_fieldspec)
+            query = qfparser.append_filter(query, filter)
+        async with graph_ctx.db.begin_readonly() as conn:
+            result = await conn.execute(query)
+            return result.scalar()
+
+    @classmethod
+    async def load_slice_project(
+        cls,
+        graph_ctx: GraphQueryContext,
+        limit: int,
+        offset: int,
+        *,
+        domain_name: str = None,
+        group_id: uuid.UUID = None,
+        user_id: uuid.UUID = None,
+        filter: str = None,
+        order: str = None,
+    ) -> Sequence[VirtualFolder]:
+        from ai.backend.manager.models import association_groups_users as agus
+
+        from .group import groups
+        from .user import users
+
+        j = sa.join(agus, users, agus.c.user_id == users.c.uuid)
+        query = sa.select([agus.c.group_id]).select_from(j).where(agus.c.user_id == user_id)
+        async with graph_ctx.db.begin_readonly() as conn:
+            result = await conn.execute(query)
+        grps = result.fetchall()
+        group_ids = [g.group_id for g in grps]
+        j = vfolders.join(groups, vfolders.c.group == groups.c.id)
+        query = (
+            sa.select([vfolders, users.c.email, groups.c.name.label("groups_name")])
```

Comment:
```suggestion
            sa.select([vfolders, groups.c.name.label("groups_name")])
```
We cannot fetch `users.c.email` column here because `group` owned vfolders have `null` value on `user` column.
Instead, we can use vfolder's `creator` field here.

Reply:
Currently, `load_slice_project`'s return value is related to class method `from_row`.
In `from_row`, `users_email` is required. 

>   File "/Users/choijiwon/bai-dev/src/ai/backend/manager/models/vfolder.py", line 1031, in from_row
>   user_email=row["users_email"],
>              ~~~^^^^^^^^^^^^^^^

So, I think we have to leave `users.c.email` as before.

Reply:
This will return the requester's email or return no group-owned folders or just generate cartesian products that we must avoid.
So we should fix `from_row()` or test by replacing `users.c.email` to `vfolders.c.creator.label("users_email")` instead of leaving this part.
Could you test `vfolders.c.creator.label("users_email")` first?